numpy/lib/tests/test_twodim_base.py

```diff
@@ -199,6 +270,33 @@ def test_axes(self):
         a = ones((50, 40, 3))
         assert_equal(rot90(a).shape, (40, 50, 3))
 
+    def test_rotation_axes(self):
+        a = np.arange(8).reshape((2,2,2))
+
+        a_rot90_01 = [[[2, 3],
+                       [6, 7]],
+                      [[0, 1],
+                       [4, 5]]]
+        a_rot90_12 = [[[1, 3],
+                       [0, 2]],
+                      [[5, 7],
+                       [4, 6]]]
+        a_rot90_20 = [[[4, 0],
+                       [6, 2]],
+                      [[5, 1],
+                       [7, 3]]]
+        a_rot90_10 = [[[4, 5],
+                       [0, 1]],
+                      [[6, 7],
+                       [2, 3]]]
+
+        assert_equal(rot90(a, axes=(0, 1)), a_rot90_01)
+        assert_equal(rot90(a, axes=(1, 0)), a_rot90_10)
+        assert_equal(rot90(a, axes=(1, 2)), a_rot90_12)
+
+        for k in range(1,5):
+            assert_equal(rot90(a, k=k, axes=(2, 0)), 
+                    rot90(a_rot90_20, k=k-1, axes=(2, 0)))
```

Comment:
added raise of ValueError and an assert_test in [1a52123](https://github.com/erensezener/numpy/commit/1a521230bc2ccce71e3bbee722364206eb090aa9)